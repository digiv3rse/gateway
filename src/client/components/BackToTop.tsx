import React from 'react';
import { css } from '@emotion/react';
import {
  background,
  brandText,
  brandAlt,
} from '@guardian/src-foundations/palette';
import { Link } from '@guardian/src-link';
import { textSans } from '@guardian/src-foundations/typography';
import { space } from '@guardian/src-foundations';

const iconContainer = css`
  position: relative;
  float: right;
  border-radius: 100%;
  background-color: ${background.primary};
  cursor: pointer;
  height: 42px;
  min-width: 42px;
`;

const linkStyles = css`
  text-decoration: none;
  color: ${brandText.anchorPrimary};
  font-weight: bold;
  line-height: 42px;
  display: flex;
  flex-direction: row;
  width: 133px;

  :hover {
    color: ${brandAlt[400]};

    .back-to-top-icon {
      background-color: ${brandAlt[400]};
    }
  }
`;

const icon = css`
  :before {
    position: absolute;
    top: 6px;
    bottom: 0;
    left: 0;
    right: 0;
    margin: auto;
    border: 2px solid ${brandText.ctaPrimary};
    border-bottom: 0;
    border-right: 0;
    content: '';
    height: 12px;
    width: 12px;
    transform: rotate(45deg);
  }
`;

const textStyles = css`
  display: flex;
  align-items: center;
  ${textSans.small({ fontWeight: 'bold' })};
  margin-right: ${space[2]}px;
`;

export const BackToTop: React.FC = () => (
  <Link cssOverrides={linkStyles} href="#top">
    <span css={textStyles}>Back to top</span>
    <span className="back-to-top-icon" css={iconContainer}>
      <i css={icon} />
    </span>
  </Link>
);
