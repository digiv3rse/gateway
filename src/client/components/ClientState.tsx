import React, { createContext, FunctionComponent } from 'react';
import { ClientState } from '@/shared/model/ClientState';

export const ClientStateContext = createContext({
  clientHosts: { idapiBaseUrl: '', oauthBaseUrl: '' },
  recaptchaConfig: { recaptchaSiteKey: '' },
});

type ClientStateProps = {
  clientState: ClientState;
};

export const ClientStateProvider: FunctionComponent<ClientStateProps> = (
  props,
) => {
  return (
    <ClientStateContext.Provider value={props.clientState}>
      {props.children}
    </ClientStateContext.Provider>
  );
};
