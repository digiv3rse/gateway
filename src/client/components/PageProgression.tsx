import React, { FunctionComponent } from 'react';
import { css } from '@emotion/core';
import { brand, neutral } from '@guardian/src-foundations/palette';
import { textSans } from '@guardian/src-foundations/typography';
import { SvgCheckmark } from '@guardian/src-icons';

const BORDER_SIZE = 2;
const CIRCLE_DIAMETER = 24;

const ul = css`
  display: flex;
  list-style: none;
  height: 54px;
  padding: 0;
  margin: 0;
`;

const circle = `
  content: ' ';
  box-sizing: content-box;
  border-radius: 50%;
  height: ${CIRCLE_DIAMETER}px;
  position: absolute;
  left: 0;
  top: 0;
  width: ${CIRCLE_DIAMETER}px;
`;

const li = (numPages: number) => css`
  ${textSans.small()}
  position: relative;
  width: ${100 / numPages}%;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  &.active {
    ${textSans.small({ fontWeight: 'bold' })}
  }
  &::after,
  &.complete::after {
    content: ' ';
    background-color: ${neutral[60]};
    height: ${BORDER_SIZE}px;
    position: absolute;
    // Border position from top is distance of a semicircle minus half the border thickness
    top: ${CIRCLE_DIAMETER / 2 + BORDER_SIZE - BORDER_SIZE / 2}px;
    left: ${CIRCLE_DIAMETER + 2 * BORDER_SIZE}px;
    right: 0;
  }
  &.complete::after {
    height: ${BORDER_SIZE * 2}px;
    background-color: ${brand[400]};
  }
  &:last-child::after {
    display: none;
  }
  &::before {
    border: ${BORDER_SIZE}px solid ${neutral[60]};
    border-radius: 50%;
    ${circle}
  }
  &.active::before,
  &.complete::before {
    content: ' ';
    background-color: ${brand[400]};
    border: ${BORDER_SIZE}px solid ${brand[400]};
    ${circle}
  }
  & svg {
    display: none;
  }
  &.complete svg {
    position: absolute;
    display: block;
    stroke: white;
    fill: white;
    height: ${CIRCLE_DIAMETER}px;
    width: ${CIRCLE_DIAMETER}px;
    margin: ${BORDER_SIZE}px;
    top: 0;
    left: 0;
    z-index: 1;
  }
`;

interface PageProgressionProps {
  pages: string[];
  current?: string;
}

export const PageProgression: FunctionComponent<PageProgressionProps> = ({
  pages,
  current,
}) => {
  const active = current ? pages.indexOf(current) : 0;
  const getClassName = (i: number) => {
    switch (true) {
      case i === active:
        return 'active';
      case i < active:
        return 'complete';
      default:
        return '';
    }
  };
  return (
    <ul css={ul}>
      {pages.map((page, i) => (
        <li className={getClassName(i)} key={i} css={li(pages.length)}>
          <SvgCheckmark />
          <div>{page}</div>
        </li>
      ))}
    </ul>
  );
};
