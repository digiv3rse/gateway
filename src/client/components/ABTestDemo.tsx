import React, { useContext } from 'react';
import { useAB } from '@guardian/ab-react';
import { tests } from '@/shared/model/experiments/abTests';
import { abTestTest } from '@/shared/model/experiments/tests/ab-test-test';
import { ClientState } from '@/shared/model/ClientState';
import { ClientStateContext } from './ClientState';

export const ABTestDemo = () => {
  // load the AB Hook
  const ABTestAPI = useAB();

  // WAYS TO RUN AB TESTS
  // 1) Using the API (recommended)
  // More documentation at https://github.com/guardian/ab-testing#the-api

  // A) Example for getting and running the first possible runnable test:
  // Get the first possible runnable test
  const firstRunnableTest = ABTestAPI.firstRunnableTest(tests);
  // Get the variant information to run
  const variantFromRunnable = firstRunnableTest?.variantToRun;
  // Get the test method which should run
  const testToRun = variantFromRunnable?.test;
  // Run the test method
  console.log(
    'A) API - First Runnable Test - Outcome:',
    testToRun && testToRun({}),
  );

  // B) Example for checking if a user is in a particular test
  // then get the variant, and run the test method
  const runnableTest = ABTestAPI.runnableTest(abTestTest);
  console.log(
    'B) API - Check for Test - Outcome:',
    runnableTest?.variantToRun.test({}),
  );

  // C) Example for checking if a user is in a specific test and variant
  const isUserInVariant = ABTestAPI.isUserInVariant(
    abTestTest.id,
    abTestTest.variants[0].id,
  );
  console.log(
    'C) API - Check for test and variant boolean - Outcome:',
    isUserInVariant,
  );

  // you can use any of the above for conditional logic too
  // example using C)
  if (isUserInVariant) {
    console.log('C) User in a variant');
  } else {
    console.log('C) User not in a variant');
  }

  // 2) Using the ClientState (not recommended)
  // In the ClientState, we pass the testId and variant of any tests
  // the user is in, you can check this too.
  // This isn't recommended as the ClientState isn't guaranteed to be available
  // It also only has the test id and variant
  // so may be more complex to run a test using this

  // Example:
  // Load the client state
  const clientState: ClientState = useContext(ClientStateContext);
  // Get the AB Tests from clientState, these are the AB Tests the user is in with
  // format is { [key: string]: { variant: string }}
  const { abTesting: { participations = {} } = {} } = clientState;
  // check if user in a test
  if (participations[abTestTest.id]?.variant === abTestTest.variants[0].id) {
    console.log(
      `2) I'm in the "${abTestTest.id}" test and "${abTestTest.variants[0].id}" variant.`,
    );
  }

  // Why AB Test?
  // In react you could return a different Component:
  if (isUserInVariant) {
    return (
      <h1>
        AB TEST DEMO - In the &quot;{abTestTest.id}&quot; test and the &quot;
        {abTestTest.variants[0].id}&quot; variant.
      </h1>
    );
  } else {
    return (
      <h1>
        AB TEST DEMO - NOT in the &quot;{abTestTest.id}&quot; test and the
        &quot;
        {abTestTest.variants[0].id}&quot; variant.
      </h1>
    );
  }
};
