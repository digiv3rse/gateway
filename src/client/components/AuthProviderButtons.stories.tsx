import React from 'react';
import { Meta } from '@storybook/react';

import { AuthProviderButtons } from '@/client/components/AuthProviderButtons';

export default {
	title: 'Components/AuthProviderButtons',
	component: AuthProviderButtons,
} as Meta;

export const Desktop = () => (
	<AuthProviderButtons
		queryParams={{ returnUrl: 'https://www.theguardian.com/uk/' }}
		providers={['social']}
	/>
);
Desktop.storyName = 'At desktop';
Desktop.parameters = {
	viewport: {
		defaultViewport: 'DESKTOP',
	},
	chromatic: { viewports: [1300] },
};

export const Mobile = () => (
	<AuthProviderButtons
		queryParams={{ returnUrl: 'https://www.theguardian.com/uk/' }}
		providers={['social']}
	/>
);
Mobile.storyName = 'At mobile 320';
Mobile.parameters = {
	viewport: {
		defaultViewport: 'MOBILE_320',
	},
	chromatic: { viewports: [320] },
};

export const NativeAppAndroid = () => (
	<AuthProviderButtons
		queryParams={{ returnUrl: 'https://www.theguardian.com/uk/' }}
		isNativeApp="android"
		providers={['social']}
	/>
);
NativeAppAndroid.storyName = 'Android native app';

export const NativeAppIos = () => (
	<AuthProviderButtons
		queryParams={{ returnUrl: 'https://www.theguardian.com/uk/' }}
		isNativeApp="ios"
		providers={['social']}
	/>
);
NativeAppIos.storyName = 'iOS native app';

export const DesktopWithEmail = () => (
	<AuthProviderButtons
		queryParams={{ returnUrl: 'https://www.theguardian.com/uk/' }}
		providers={['social', 'email']}
	/>
);
DesktopWithEmail.storyName = 'At desktop (with email)';
DesktopWithEmail.parameters = {
	viewport: {
		defaultViewport: 'DESKTOP',
	},
	chromatic: { viewports: [1300] },
};

export const MobileWithEmail = () => (
	<AuthProviderButtons
		queryParams={{ returnUrl: 'https://www.theguardian.com/uk/' }}
		providers={['social', 'email']}
	/>
);
MobileWithEmail.storyName = 'At mobile 320 (with email)';
MobileWithEmail.parameters = {
	viewport: {
		defaultViewport: 'MOBILE_320',
	},
	chromatic: { viewports: [320] },
};

export const NativeAppAndroidWithEmail = () => (
	<AuthProviderButtons
		queryParams={{ returnUrl: 'https://www.theguardian.com/uk/' }}
		isNativeApp="android"
		providers={['social', 'email']}
	/>
);
NativeAppAndroidWithEmail.storyName = 'Android native app (with email)';

export const NativeAppIosWithEmail = () => (
	<AuthProviderButtons
		queryParams={{ returnUrl: 'https://www.theguardian.com/uk/' }}
		isNativeApp="ios"
		providers={['social', 'email']}
	/>
);
NativeAppIosWithEmail.storyName = 'iOS native app (with email)';
