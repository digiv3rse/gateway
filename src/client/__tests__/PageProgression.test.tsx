import React from 'react';
import { PageProgression } from '@/client/components/PageProgression';
import { create } from 'react-test-renderer';

describe('PageProgression component', () => {
  describe('when no active page is set', () => {
    it('shows the first step as the active step', () => {
      const rendered = create(<PageProgression />);
      expect(rendered.toJSON()).toMatchSnapshot();
    });
  });
  describe('when an active section is supplied', () => {
    it('marks the previous sections as done and the current one as active', () => {
      const rendered = create(<PageProgression current={'Newsletters'} />);
      expect(rendered.toJSON()).toMatchSnapshot();
    });
  });
  describe('when on the last section', () => {
    it('marks the previous sections as done and the last one as active', () => {
      const rendered = create(<PageProgression current={'Review'} />);
      expect(rendered.toJSON()).toMatchSnapshot();
    });
  });
});
