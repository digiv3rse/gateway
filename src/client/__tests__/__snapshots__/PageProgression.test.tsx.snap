// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageProgression component when an active section is supplied marks the previous sections as done and the current one as active 1`] = `
<ul
  css={
    Object {
      "map": undefined,
      "name": "48yusx",
      "next": undefined,
      "styles": "
  display: flex;
  list-style: none;
  height: 54px;
  padding: 0;
  margin: 0;
",
      "toString": [Function],
    }
  }
>
  <li
    className="active"
    css={
      Object {
        "map": undefined,
        "name": "1g4ulkr",
        "next": undefined,
        "styles": "
  
	font-family: GuardianTextSans, Guardian Text Sans Web, Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif;
	font-size: 0.9375rem;
	line-height: 1.5;
	font-weight: 400;
	;

  position: relative;
  width: 33.333333333333336%;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  &.active {
    
	font-family: GuardianTextSans, Guardian Text Sans Web, Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif;
	font-size: 0.9375rem;
	line-height: 1.5;
	font-weight: 700;
	;

  }
  &::after,
  &.complete::after {
    content: ' ';
    background-color: #999999;
    height: 2px;
    position: absolute;
    // Border position from top is distance of a semicircle minus half the border thickness
    top: 13px;
    left: 28px;
    right: 0;
  }
  &.complete::after {
    height: 4px;
    background-color: #052962;
  }
  &:last-child::after {
    display: none;
  }
  &::before {
    border: 2px solid #999999;
    border-radius: 50%;
    
  content: ' ';
  box-sizing: content-box;
  border-radius: 50%;
  height: 24px;
  position: absolute;
  left: 0;
  top: 0;
  width: 24px;

  }
  &.active::before,
  &.complete::before {
    content: ' ';
    background-color: #052962;
    border: 2px solid #052962;
    
  content: ' ';
  box-sizing: content-box;
  border-radius: 50%;
  height: 24px;
  position: absolute;
  left: 0;
  top: 0;
  width: 24px;

  }
  & svg {
    display: none;
  }
  &.complete svg {
    position: absolute;
    display: block;
    stroke: white;
    fill: white;
    height: 24px;
    width: 24px;
    margin: 2px;
    top: 0;
    left: 0;
    z-index: 1;
  }
",
        "toString": [Function],
      }
    }
  >
    <svg
      viewBox="0 0 30 30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M6 14.775l-1 1 5 7h.475l14.3-14.8-1-.975-13.3 12.05L6 14.775z"
        fillRule="evenodd"
      />
    </svg>
    <div>
      Newsletters
    </div>
  </li>
  <li
    className=""
    css={
      Object {
        "map": undefined,
        "name": "1g4ulkr",
        "next": undefined,
        "styles": "
  
	font-family: GuardianTextSans, Guardian Text Sans Web, Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif;
	font-size: 0.9375rem;
	line-height: 1.5;
	font-weight: 400;
	;

  position: relative;
  width: 33.333333333333336%;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  &.active {
    
	font-family: GuardianTextSans, Guardian Text Sans Web, Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif;
	font-size: 0.9375rem;
	line-height: 1.5;
	font-weight: 700;
	;

  }
  &::after,
  &.complete::after {
    content: ' ';
    background-color: #999999;
    height: 2px;
    position: absolute;
    // Border position from top is distance of a semicircle minus half the border thickness
    top: 13px;
    left: 28px;
    right: 0;
  }
  &.complete::after {
    height: 4px;
    background-color: #052962;
  }
  &:last-child::after {
    display: none;
  }
  &::before {
    border: 2px solid #999999;
    border-radius: 50%;
    
  content: ' ';
  box-sizing: content-box;
  border-radius: 50%;
  height: 24px;
  position: absolute;
  left: 0;
  top: 0;
  width: 24px;

  }
  &.active::before,
  &.complete::before {
    content: ' ';
    background-color: #052962;
    border: 2px solid #052962;
    
  content: ' ';
  box-sizing: content-box;
  border-radius: 50%;
  height: 24px;
  position: absolute;
  left: 0;
  top: 0;
  width: 24px;

  }
  & svg {
    display: none;
  }
  &.complete svg {
    position: absolute;
    display: block;
    stroke: white;
    fill: white;
    height: 24px;
    width: 24px;
    margin: 2px;
    top: 0;
    left: 0;
    z-index: 1;
  }
",
        "toString": [Function],
      }
    }
  >
    <svg
      viewBox="0 0 30 30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M6 14.775l-1 1 5 7h.475l14.3-14.8-1-.975-13.3 12.05L6 14.775z"
        fillRule="evenodd"
      />
    </svg>
    <div>
      Stay in touch
    </div>
  </li>
  <li
    className=""
    css={
      Object {
        "map": undefined,
        "name": "1g4ulkr",
        "next": undefined,
        "styles": "
  
	font-family: GuardianTextSans, Guardian Text Sans Web, Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif;
	font-size: 0.9375rem;
	line-height: 1.5;
	font-weight: 400;
	;

  position: relative;
  width: 33.333333333333336%;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  &.active {
    
	font-family: GuardianTextSans, Guardian Text Sans Web, Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif;
	font-size: 0.9375rem;
	line-height: 1.5;
	font-weight: 700;
	;

  }
  &::after,
  &.complete::after {
    content: ' ';
    background-color: #999999;
    height: 2px;
    position: absolute;
    // Border position from top is distance of a semicircle minus half the border thickness
    top: 13px;
    left: 28px;
    right: 0;
  }
  &.complete::after {
    height: 4px;
    background-color: #052962;
  }
  &:last-child::after {
    display: none;
  }
  &::before {
    border: 2px solid #999999;
    border-radius: 50%;
    
  content: ' ';
  box-sizing: content-box;
  border-radius: 50%;
  height: 24px;
  position: absolute;
  left: 0;
  top: 0;
  width: 24px;

  }
  &.active::before,
  &.complete::before {
    content: ' ';
    background-color: #052962;
    border: 2px solid #052962;
    
  content: ' ';
  box-sizing: content-box;
  border-radius: 50%;
  height: 24px;
  position: absolute;
  left: 0;
  top: 0;
  width: 24px;

  }
  & svg {
    display: none;
  }
  &.complete svg {
    position: absolute;
    display: block;
    stroke: white;
    fill: white;
    height: 24px;
    width: 24px;
    margin: 2px;
    top: 0;
    left: 0;
    z-index: 1;
  }
",
        "toString": [Function],
      }
    }
  >
    <svg
      viewBox="0 0 30 30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M6 14.775l-1 1 5 7h.475l14.3-14.8-1-.975-13.3 12.05L6 14.775z"
        fillRule="evenodd"
      />
    </svg>
    <div>
      Your data
    </div>
  </li>
</ul>
`;

exports[`PageProgression component when no active page is set shows the first step as the active step 1`] = `
<ul
  css={
    Object {
      "map": undefined,
      "name": "48yusx",
      "next": undefined,
      "styles": "
  display: flex;
  list-style: none;
  height: 54px;
  padding: 0;
  margin: 0;
",
      "toString": [Function],
    }
  }
>
  <li
    className="active"
    css={
      Object {
        "map": undefined,
        "name": "1g4ulkr",
        "next": undefined,
        "styles": "
  
	font-family: GuardianTextSans, Guardian Text Sans Web, Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif;
	font-size: 0.9375rem;
	line-height: 1.5;
	font-weight: 400;
	;

  position: relative;
  width: 33.333333333333336%;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  &.active {
    
	font-family: GuardianTextSans, Guardian Text Sans Web, Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif;
	font-size: 0.9375rem;
	line-height: 1.5;
	font-weight: 700;
	;

  }
  &::after,
  &.complete::after {
    content: ' ';
    background-color: #999999;
    height: 2px;
    position: absolute;
    // Border position from top is distance of a semicircle minus half the border thickness
    top: 13px;
    left: 28px;
    right: 0;
  }
  &.complete::after {
    height: 4px;
    background-color: #052962;
  }
  &:last-child::after {
    display: none;
  }
  &::before {
    border: 2px solid #999999;
    border-radius: 50%;
    
  content: ' ';
  box-sizing: content-box;
  border-radius: 50%;
  height: 24px;
  position: absolute;
  left: 0;
  top: 0;
  width: 24px;

  }
  &.active::before,
  &.complete::before {
    content: ' ';
    background-color: #052962;
    border: 2px solid #052962;
    
  content: ' ';
  box-sizing: content-box;
  border-radius: 50%;
  height: 24px;
  position: absolute;
  left: 0;
  top: 0;
  width: 24px;

  }
  & svg {
    display: none;
  }
  &.complete svg {
    position: absolute;
    display: block;
    stroke: white;
    fill: white;
    height: 24px;
    width: 24px;
    margin: 2px;
    top: 0;
    left: 0;
    z-index: 1;
  }
",
        "toString": [Function],
      }
    }
  >
    <svg
      viewBox="0 0 30 30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M6 14.775l-1 1 5 7h.475l14.3-14.8-1-.975-13.3 12.05L6 14.775z"
        fillRule="evenodd"
      />
    </svg>
    <div>
      Newsletters
    </div>
  </li>
  <li
    className=""
    css={
      Object {
        "map": undefined,
        "name": "1g4ulkr",
        "next": undefined,
        "styles": "
  
	font-family: GuardianTextSans, Guardian Text Sans Web, Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif;
	font-size: 0.9375rem;
	line-height: 1.5;
	font-weight: 400;
	;

  position: relative;
  width: 33.333333333333336%;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  &.active {
    
	font-family: GuardianTextSans, Guardian Text Sans Web, Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif;
	font-size: 0.9375rem;
	line-height: 1.5;
	font-weight: 700;
	;

  }
  &::after,
  &.complete::after {
    content: ' ';
    background-color: #999999;
    height: 2px;
    position: absolute;
    // Border position from top is distance of a semicircle minus half the border thickness
    top: 13px;
    left: 28px;
    right: 0;
  }
  &.complete::after {
    height: 4px;
    background-color: #052962;
  }
  &:last-child::after {
    display: none;
  }
  &::before {
    border: 2px solid #999999;
    border-radius: 50%;
    
  content: ' ';
  box-sizing: content-box;
  border-radius: 50%;
  height: 24px;
  position: absolute;
  left: 0;
  top: 0;
  width: 24px;

  }
  &.active::before,
  &.complete::before {
    content: ' ';
    background-color: #052962;
    border: 2px solid #052962;
    
  content: ' ';
  box-sizing: content-box;
  border-radius: 50%;
  height: 24px;
  position: absolute;
  left: 0;
  top: 0;
  width: 24px;

  }
  & svg {
    display: none;
  }
  &.complete svg {
    position: absolute;
    display: block;
    stroke: white;
    fill: white;
    height: 24px;
    width: 24px;
    margin: 2px;
    top: 0;
    left: 0;
    z-index: 1;
  }
",
        "toString": [Function],
      }
    }
  >
    <svg
      viewBox="0 0 30 30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M6 14.775l-1 1 5 7h.475l14.3-14.8-1-.975-13.3 12.05L6 14.775z"
        fillRule="evenodd"
      />
    </svg>
    <div>
      Stay in touch
    </div>
  </li>
  <li
    className=""
    css={
      Object {
        "map": undefined,
        "name": "1g4ulkr",
        "next": undefined,
        "styles": "
  
	font-family: GuardianTextSans, Guardian Text Sans Web, Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif;
	font-size: 0.9375rem;
	line-height: 1.5;
	font-weight: 400;
	;

  position: relative;
  width: 33.333333333333336%;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  &.active {
    
	font-family: GuardianTextSans, Guardian Text Sans Web, Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif;
	font-size: 0.9375rem;
	line-height: 1.5;
	font-weight: 700;
	;

  }
  &::after,
  &.complete::after {
    content: ' ';
    background-color: #999999;
    height: 2px;
    position: absolute;
    // Border position from top is distance of a semicircle minus half the border thickness
    top: 13px;
    left: 28px;
    right: 0;
  }
  &.complete::after {
    height: 4px;
    background-color: #052962;
  }
  &:last-child::after {
    display: none;
  }
  &::before {
    border: 2px solid #999999;
    border-radius: 50%;
    
  content: ' ';
  box-sizing: content-box;
  border-radius: 50%;
  height: 24px;
  position: absolute;
  left: 0;
  top: 0;
  width: 24px;

  }
  &.active::before,
  &.complete::before {
    content: ' ';
    background-color: #052962;
    border: 2px solid #052962;
    
  content: ' ';
  box-sizing: content-box;
  border-radius: 50%;
  height: 24px;
  position: absolute;
  left: 0;
  top: 0;
  width: 24px;

  }
  & svg {
    display: none;
  }
  &.complete svg {
    position: absolute;
    display: block;
    stroke: white;
    fill: white;
    height: 24px;
    width: 24px;
    margin: 2px;
    top: 0;
    left: 0;
    z-index: 1;
  }
",
        "toString": [Function],
      }
    }
  >
    <svg
      viewBox="0 0 30 30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M6 14.775l-1 1 5 7h.475l14.3-14.8-1-.975-13.3 12.05L6 14.775z"
        fillRule="evenodd"
      />
    </svg>
    <div>
      Your data
    </div>
  </li>
</ul>
`;

exports[`PageProgression component when on the last section marks the previous sections as done and the last one as active 1`] = `
<ul
  css={
    Object {
      "map": undefined,
      "name": "48yusx",
      "next": undefined,
      "styles": "
  display: flex;
  list-style: none;
  height: 54px;
  padding: 0;
  margin: 0;
",
      "toString": [Function],
    }
  }
>
  <li
    className=""
    css={
      Object {
        "map": undefined,
        "name": "1g4ulkr",
        "next": undefined,
        "styles": "
  
	font-family: GuardianTextSans, Guardian Text Sans Web, Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif;
	font-size: 0.9375rem;
	line-height: 1.5;
	font-weight: 400;
	;

  position: relative;
  width: 33.333333333333336%;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  &.active {
    
	font-family: GuardianTextSans, Guardian Text Sans Web, Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif;
	font-size: 0.9375rem;
	line-height: 1.5;
	font-weight: 700;
	;

  }
  &::after,
  &.complete::after {
    content: ' ';
    background-color: #999999;
    height: 2px;
    position: absolute;
    // Border position from top is distance of a semicircle minus half the border thickness
    top: 13px;
    left: 28px;
    right: 0;
  }
  &.complete::after {
    height: 4px;
    background-color: #052962;
  }
  &:last-child::after {
    display: none;
  }
  &::before {
    border: 2px solid #999999;
    border-radius: 50%;
    
  content: ' ';
  box-sizing: content-box;
  border-radius: 50%;
  height: 24px;
  position: absolute;
  left: 0;
  top: 0;
  width: 24px;

  }
  &.active::before,
  &.complete::before {
    content: ' ';
    background-color: #052962;
    border: 2px solid #052962;
    
  content: ' ';
  box-sizing: content-box;
  border-radius: 50%;
  height: 24px;
  position: absolute;
  left: 0;
  top: 0;
  width: 24px;

  }
  & svg {
    display: none;
  }
  &.complete svg {
    position: absolute;
    display: block;
    stroke: white;
    fill: white;
    height: 24px;
    width: 24px;
    margin: 2px;
    top: 0;
    left: 0;
    z-index: 1;
  }
",
        "toString": [Function],
      }
    }
  >
    <svg
      viewBox="0 0 30 30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M6 14.775l-1 1 5 7h.475l14.3-14.8-1-.975-13.3 12.05L6 14.775z"
        fillRule="evenodd"
      />
    </svg>
    <div>
      Newsletters
    </div>
  </li>
  <li
    className=""
    css={
      Object {
        "map": undefined,
        "name": "1g4ulkr",
        "next": undefined,
        "styles": "
  
	font-family: GuardianTextSans, Guardian Text Sans Web, Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif;
	font-size: 0.9375rem;
	line-height: 1.5;
	font-weight: 400;
	;

  position: relative;
  width: 33.333333333333336%;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  &.active {
    
	font-family: GuardianTextSans, Guardian Text Sans Web, Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif;
	font-size: 0.9375rem;
	line-height: 1.5;
	font-weight: 700;
	;

  }
  &::after,
  &.complete::after {
    content: ' ';
    background-color: #999999;
    height: 2px;
    position: absolute;
    // Border position from top is distance of a semicircle minus half the border thickness
    top: 13px;
    left: 28px;
    right: 0;
  }
  &.complete::after {
    height: 4px;
    background-color: #052962;
  }
  &:last-child::after {
    display: none;
  }
  &::before {
    border: 2px solid #999999;
    border-radius: 50%;
    
  content: ' ';
  box-sizing: content-box;
  border-radius: 50%;
  height: 24px;
  position: absolute;
  left: 0;
  top: 0;
  width: 24px;

  }
  &.active::before,
  &.complete::before {
    content: ' ';
    background-color: #052962;
    border: 2px solid #052962;
    
  content: ' ';
  box-sizing: content-box;
  border-radius: 50%;
  height: 24px;
  position: absolute;
  left: 0;
  top: 0;
  width: 24px;

  }
  & svg {
    display: none;
  }
  &.complete svg {
    position: absolute;
    display: block;
    stroke: white;
    fill: white;
    height: 24px;
    width: 24px;
    margin: 2px;
    top: 0;
    left: 0;
    z-index: 1;
  }
",
        "toString": [Function],
      }
    }
  >
    <svg
      viewBox="0 0 30 30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M6 14.775l-1 1 5 7h.475l14.3-14.8-1-.975-13.3 12.05L6 14.775z"
        fillRule="evenodd"
      />
    </svg>
    <div>
      Stay in touch
    </div>
  </li>
  <li
    className=""
    css={
      Object {
        "map": undefined,
        "name": "1g4ulkr",
        "next": undefined,
        "styles": "
  
	font-family: GuardianTextSans, Guardian Text Sans Web, Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif;
	font-size: 0.9375rem;
	line-height: 1.5;
	font-weight: 400;
	;

  position: relative;
  width: 33.333333333333336%;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  &.active {
    
	font-family: GuardianTextSans, Guardian Text Sans Web, Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif;
	font-size: 0.9375rem;
	line-height: 1.5;
	font-weight: 700;
	;

  }
  &::after,
  &.complete::after {
    content: ' ';
    background-color: #999999;
    height: 2px;
    position: absolute;
    // Border position from top is distance of a semicircle minus half the border thickness
    top: 13px;
    left: 28px;
    right: 0;
  }
  &.complete::after {
    height: 4px;
    background-color: #052962;
  }
  &:last-child::after {
    display: none;
  }
  &::before {
    border: 2px solid #999999;
    border-radius: 50%;
    
  content: ' ';
  box-sizing: content-box;
  border-radius: 50%;
  height: 24px;
  position: absolute;
  left: 0;
  top: 0;
  width: 24px;

  }
  &.active::before,
  &.complete::before {
    content: ' ';
    background-color: #052962;
    border: 2px solid #052962;
    
  content: ' ';
  box-sizing: content-box;
  border-radius: 50%;
  height: 24px;
  position: absolute;
  left: 0;
  top: 0;
  width: 24px;

  }
  & svg {
    display: none;
  }
  &.complete svg {
    position: absolute;
    display: block;
    stroke: white;
    fill: white;
    height: 24px;
    width: 24px;
    margin: 2px;
    top: 0;
    left: 0;
    z-index: 1;
  }
",
        "toString": [Function],
      }
    }
  >
    <svg
      viewBox="0 0 30 30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M6 14.775l-1 1 5 7h.475l14.3-14.8-1-.975-13.3 12.05L6 14.775z"
        fillRule="evenodd"
      />
    </svg>
    <div>
      Your data
    </div>
  </li>
</ul>
`;
